client/internal/networkmonitor/monitor_bsd.go

```diff
@@ -21,19 +22,28 @@ func checkChange(ctx context.Context, nexthopv4, nexthopv6 systemops.Nexthop, ca
 		return fmt.Errorf("failed to open routing socket: %v", err)
 	}
 	defer func() {
-		if err := unix.Close(fd); err != nil {
+		err := unix.Close(fd)
+		if err != nil && !errors.Is(err, unix.EBADF) {
 			log.Errorf("Network monitor: failed to close routing socket: %v", err)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		err := unix.Close(fd)
+		if err != nil && !errors.Is(err, unix.EBADF) {
+			log.Debugf("Network monitor: closed routing socket")
+		}
+	}()
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ErrStopped
 		default:
 			buf := make([]byte, 2048)
 			n, err := unix.Read(fd, buf)
-			if err != nil {
+			if err != nil && !errors.Is(err, unix.EBADF) && !errors.Is(err, unix.EINVAL) {
```

Comment:
I think we need to `continue` on the ignored errors as well, just don't log them, or leg them on debug. Apologies for the previous suggestion

Reply:
fixed